### PR TITLE
Refactor Foundation.Commerce

### PR DESCRIPTION
**Pull request by yuya-yuliya**
- move Result, VoidResult, BaseController to Foundation.Base
- rename Pipelines/EndVisit to Pipelines/Logout
- refactor CustomerProvider

### CODE REVIEW

File: src/Foundation/Commerce/code/Providers/CustomerProvider.cs [hunk 1]
```diff
@@ -87,5 +94,14 @@ private User MapToUser(CommerceUser commerceUser, string contactIdOrEmail)
 
             return this.userMapper.Map<CommerceUser, User>(commerceUser);
         }
+
+        private User GetCommerceUser(string contactIdOrName)
```
Comment (larionovadaria) [inline src/Foundation/Commerce/code/Providers/CustomerProvider.cs]: Obsolete?

File: src/Foundation/Base/code/Models/Result.cs [hunk 2]
```diff
@@ -58,22 +56,6 @@ public void SetError(string error)
             this.Errors.Add(error);
         }
 
-        [Obsolete("This method is obsolete. Use SetErrors(IList<string> errors) instead.")]
-        public void SetErrors(ServiceProviderResult result)
```
Comment (larionovadaria) [inline src/Foundation/Base/code/Models/Result.cs]: This one is used in several commerce repositories

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/Foundation/Commerce/code/Providers/CustomerProvider.cs [hunk 3]
```diff
@@ -44,13 +45,19 @@ public CustomerProvider(IAccountManager accountManager, IUserMapper userMapper)
             this.userMapper = userMapper;
         }
 
-        public User GetCommerceUser(string contactIdOrName)
+        public User GetUser(string email)
         {
-            Assert.ArgumentNotNullOrEmpty(contactIdOrName, nameof(contactIdOrName));
+            Assert.ArgumentNotNull(email, nameof(email));
```
Comment (stanislav-kozorez) [inline src/Foundation/Commerce/code/Providers/CustomerProvider.cs]: Why it became `ArgumentNotNull`?

File: src/Foundation/Commerce/code/Providers/CustomerProvider.cs (hunk 1 above)
Comment (yuya-yuliya) [reply src/Foundation/Commerce/code/Providers/CustomerProvider.cs]: This method used inside class

**Pull request merged**